Source/Pawnmorphs/Esoteria/PawnmorphMPCompat.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Reflection;
+
+using Verse;
+using HarmonyLib;
+using Multiplayer.API;
+
+using Pawnmorph;
+using Pawnmorph.Hediffs;
+
+namespace PawnMorpher
+{
+    [StaticConstructorOnStartup]
+    static class PawnmorphMPCompat
+    {
+        static int lastTick;
+        static int lastSeed;
+
+        static PawnmorphMPCompat()
+        {
+            if (!MP.enabled) return;
+
+            PatchRand();
+            SyncMethods();
+
+            Log.Message("Pawnmorpher :: Multiplayer Compatibility enabled");
+        }
+
+        static void PatchRand()
+        {
+            var rngMethods = new MethodBase[] {
+                // Motes
+                AccessTools.Method(typeof(IntermittentMagicSprayer), nameof(IntermittentMagicSprayer.ThrowMagicPuffUp)),
+                AccessTools.Method(typeof(IntermittentMagicSprayer), nameof(IntermittentMagicSprayer.ThrowMagicPuffDown)),
+
+                // Fixes a very rare desync called from AI, I can't find the logic here, possibly a core MP issue.
+                AccessTools.Method(typeof(SpreadingMutationComp), nameof(SpreadingMutationComp.CompPostTick)),
+            };
+
+            var harmony = new Harmony("com.pawnmorpher.mpcompat");
```

Comment:
why are you using a new Harmony instance rather then the one in PawnmorphPatches.cs?

Reply:
No harm done, it's just short lived instance. Same with the one on PawnmorphPatches, they're not static fields.